packages/ics721/src/testing/integration_tests.rs

```diff
@@ -90,68 +95,205 @@ fn no_init(
 ) {
 }
 
-#[derive(Debug)]
-struct Bech32AddressGenerator {
-    pub hrp: String,
+#[derive(Default)]
+pub struct MockAddressGenerator;
+
+impl AddressGenerator for MockAddressGenerator {
```

Comment:
biggest changes are in test, using MockAddressGenerator and MockApiBech32. In generator, predictable_contract_address() function covers instantiate2.

Also all tests had to be refactored and cleanup for better readability.